lib/spack/spack/util/timer.py

```diff
@@ -11,51 +11,103 @@
 """
 import sys
 import time
+from collections import OrderedDict, namedtuple
+from contextlib import contextmanager
+
+from llnl.util.lang import pretty_seconds
 
 import spack.util.spack_json as sjson
 
+Interval = namedtuple("Interval", ("begin", "end"))
+
 
 class Timer(object):
-    """
-    Simple timer for timing phases of a solve or install
-    """
-
-    def __init__(self):
-        self.start = time.time()
-        self.last = self.start
-        self.phases = {}
-        self.end = None
-
-    def phase(self, name):
-        last = self.last
-        now = time.time()
-        self.phases[name] = now - last
-        self.last = now
+    """Simple interval timer"""
 
-    @property
-    def total(self):
-        """Return the total time"""
-        if self.end:
-            return self.end - self.start
-        return time.time() - self.start
+    def __init__(self, now=time.time):
+        """
+        Arguments:
+            now: function that gives the seconds since e.g. epoch
+        """
+        self._now = now
+        self._timers = OrderedDict()  # type: OrderedDict[str,Interval]
+
+        # _global is the overal timer since the instance was created
+        self._timers["_global"] = Interval(self._now(), end=None)
```

Comment:
Can we reference `"_global"` in a variable instead of using literal string in multiple places?